TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at runNextTicks (node:internal/process/task_queues:64:3)
    at listOnTimeout (node:internal/timers:540:9)
    at processTimers (node:internal/timers:514:7)
    at AzureOpenAiLLM.chat (file:///home/adib/evaluator/src/util/llm/wrappers/azure.ts:42:26)
    at Chat.handleLLMChat (file:///home/adib/evaluator/src/util/llm/index.ts:228:30)
    at Chat.askChallenge (file:///home/adib/evaluator/src/util/llm/index.ts:94:25)
    at runEval (file:///home/adib/evaluator/src/eval-unit.ts:78:19)
    at file:///home/adib/evaluator/src/rum-model.ts:112:21 {
  cause: AggregateError: 
      at internalConnectMultiple (node:net:1116:18)
      at internalConnectMultiple (node:net:1184:5)
      at Timeout.internalConnectMultipleTimeout (node:net:1710:5)
      at listOnTimeout (node:internal/timers:575:11)
      at processTimers (node:internal/timers:514:7) {
    code: 'ETIMEDOUT',
    [errors]: [
      Error: connect ETIMEDOUT 172.64.150.137:443
          at createConnectionError (node:net:1646:14)
          at Timeout.internalConnectMultipleTimeout (node:net:1705:38)
          at listOnTimeout (node:internal/timers:575:11)
          at processTimers (node:internal/timers:514:7) {
        errno: -110,
        code: 'ETIMEDOUT',
        syscall: 'connect',
        address: '172.64.150.137',
        port: 443
      },
      Error: connect ENETUNREACH 2606:4700:4400::6812:2577:443 - Local (:::0)
          at internalConnectMultiple (node:net:1180:16)
          at Timeout.internalConnectMultipleTimeout (node:net:1710:5)
          at listOnTimeout (node:internal/timers:575:11)
          at processTimers (node:internal/timers:514:7) {
        errno: -101,
        code: 'ENETUNREACH',
        syscall: 'connect',
        address: '2606:4700:4400::6812:2577',
        port: 443
      },
      Error: connect ETIMEDOUT 104.18.37.119:443
          at createConnectionError (node:net:1646:14)
          at Timeout.internalConnectMultipleTimeout (node:net:1705:38)
          at listOnTimeout (node:internal/timers:575:11)
          at processTimers (node:internal/timers:514:7) {
        errno: -110,
        code: 'ETIMEDOUT',
        syscall: 'connect',
        address: '104.18.37.119',
        port: 443
      },
      Error: connect ENETUNREACH 2606:4700:4400::ac40:9689:443 - Local (:::0)
          at internalConnectMultiple (node:net:1180:16)
          at Timeout.internalConnectMultipleTimeout (node:net:1710:5)
          at listOnTimeout (node:internal/timers:575:11)
          at processTimers (node:internal/timers:514:7) {
        errno: -101,
        code: 'ENETUNREACH',
        syscall: 'connect',
        address: '2606:4700:4400::ac40:9689',
        port: 443
      }
    ]
  }
}